### ISSUE TITLE
TensorBoardOutputConfig/Sagemaker Debugger does not behave as documented

### ISSUE THREAD

**Issue by ando-khachatryan** [awslabs/sagemaker-debugger#284]
I have a working pytorch training script, which runs on my local machine. I'm trying to set it up to run on Sagemaker.
I want to save data to TensorBoard, get this uploaded into S3, and visualize via TensorBoard, like the documentation says here: https://sagemaker.readthedocs.io/en/stable/amazon_sagemaker_debugger.html#capture-real-time-tensorboard-data-from-the-debugging-hook

As mentioned, this works perfectl when running on my dev machine.
On Sagemaker, I do this: 
```
    tensorboard_output_config = TensorBoardOutputConfig(
        s3_output_path=f's3://{args.s3_bucket}/tensorboard/{args.noise}',
        container_local_output_path=f'/tb-logs'
    )
```
then run it using:
`  job = PyTorch(..., tensorboard_output_config=tensorboard_output_config, ...)`

Sagemaker creates the TensorBoard folder on S3, and seems to be logging everything I need. However, it is trying to log bunch of other things which I haven't specified. When the training starts, I see this in the log:
```
[2020-06-29 13:44:20.083 algo-1:52 INFO json_config.py:90] Creating hook from json_config at /opt/ml/input/config/debughookconfig.json.
[2020-06-29 13:44:20.084 algo-1:52 INFO hook.py:236] Saving to /opt/ml/output/tensors
[2020-06-29 13:44:20.084 algo-1:52 INFO state_store.py:67] The checkpoint config file /opt/ml/input/config/checkpointconfig.json does not exist.
[2020-06-29 13:44:20.100 algo-1:52 INFO hook.py:376] Monitoring the collections: losses
[2020-06-29 13:44:20.100 algo-1:52 INFO hook.py:437] Hook is writing from the hook with pid: 52
```

Sagemaker is trying to log the losses from my networks, because they are all added into TensorBoard, with each of the losses having one value. 
On TensorBoard, for every run, there are multiple directories and files. One of them contains all the information that I am logging myself. All the others are created my Sagemaker; they are clogging the UI and IMHO should not be there.

How can I turn this off?

**Comment by Vikas-kum** [comment]
>> On TensorBoard, for every run, there are multiple directories and files. One of them contains all the information that I am logging myself. All the others are created my Sagemaker; they are clogging the UI and IMHO should not be there.

@ando-khachatryan Would you be able to provide what directories are created and may be screenshot of UI. 
Also it will be good if you can provide a way to reproduce the problem. 

Thanks.

**Comment by ando-khachatryan** [comment]
Thanks for the prompt reply. 
Screenshots: 
S3, TensorBoard folder. 
<img width="1713" alt="Screen Shot 2020-06-29 at 21 18 01" src="https://user-images.githubusercontent.com/4376090/86036295-b4196500-ba4e-11ea-88d9-6b6bd633d3e5.png">

TensorBoard, showing the "runs". 
<img width="1783" alt="Screen Shot 2020-06-29 at 21 20 19" src="https://user-images.githubusercontent.com/4376090/86036371-ceebd980-ba4e-11ea-880d-b7a594dc2eac.png">

TensorBoard, showing the "runs" and the scalars.
<img width="1779" alt="Screen Shot 2020-06-29 at 21 32 33" src="https://user-images.githubusercontent.com/4376090/86037245-1f176b80-ba50-11ea-9cb0-898691a38ab7.png">

All the scalar groups except the last two (train and validation) are added by Sagemaker. So out of the four runs shown in TB, only the first is the info added by me. On the S3 screenshot, these are the three folders. 

Sorry for not providing a reproducible example. This is not-too-small project, but I have a git repo, I can push it and give you a small zip of the dataset. Would that work?

**Comment by Vikas-kum** [comment]
>> Sorry for not providing a reproducible example. This is not-too-small project, but I have a git repo, I can push it and give you a small zip of the dataset. Would that work?

Yes, that will help. Please provide the scripts, data and instruction to run the script.

**Comment by ando-khachatryan** [comment]
Git repo & branch: https://github.com/ando-khachatryan/HiDDeN/tree/sagemaker-debug-issue

data: https://ando-public-sagemaker.s3.amazonaws.com/small.zip
Extract and put the files into your S3 bucket.

```
% cd wm 
% cp util/sagemaker_config_example.py util/sagemaker_config.py
```
Open and fill in the Sagemaker execution ARN.
 
To run on Sagemaker, you need boto3 and sagemaker packages installed. 
`wm % python sagemaker_start.py unet-conv --noise 'jpeg()+blur()' --data-prefix <data-on-s3> --epochs 50 --wait --s3-bucket <bucket-name>
`
For instance, my data sits in data/small on my bucket, then --data-prefix is 'data/small' for me. 

Let me know if I have forgotten something. Thank you once again for your time.

**Comment by ando-khachatryan** [comment]
Update: 
I found an issue on in my code, fixed it and pushed. However the problem is not completely gone. 
<img width="1779" alt="Screen Shot 2020-07-02 at 01 29 11" src="https://user-images.githubusercontent.com/4376090/86293387-93ded700-bc03-11ea-98d0-5d92d8df828b.png">

For diagnostics, I removed all statements which write into TensorBoard. So right now I am just creating a SummaryWriter object, and that's it. It still shows a BCEWithLogitsLoss_output_0, and I have no clue where it is coming from.

**Comment by Vikas-kum** [comment]
I feel that this is not related to smdebug. IIRC, SMDebug doesn't create those s3 folder.  
Can you  confirm that you don't see those folder created locally when you run your script on local machine? 
SMDebug s3 output would look like this - 
![Screen Shot 2020-07-01 at 7 03 04 PM](https://user-images.githubusercontent.com/1817689/86308311-accc9580-bbcd-11ea-892d-909e5c9e6632.png)

**Comment by ando-khachatryan** [comment]
Yes, I ran it on the local machine and there is only the the file within the TensorBoard folder which contains the info that I write. The GLOBAL folder is not there when I run it locally. 

I also tried to remove ALL statements which write to TensorBoard, and run it on Sagemaker. This would just create the file (which does not contain anything, as expected), plus it created the GLOBAL folder. 

I checked the folders you mentioned. They are created on S3:

<img width="1762" alt="Screen Shot 2020-07-02 at 22 54 37" src="https://user-images.githubusercontent.com/4376090/86399249-fbeef500-bcb7-11ea-89f3-9f27dd99346c.png">
This is the screenshot inside the events subfolder: 

<img width="1734" alt="Screen Shot 2020-07-02 at 22 56 37" src="https://user-images.githubusercontent.com/4376090/86399377-39538280-bcb8-11ea-8228-24984c7ece04.png">

And this is the GLOBAL folder inside TensorBoard folder, the one that should not be there: 

<img width="1763" alt="Screen Shot 2020-07-02 at 22 55 27" src="https://user-images.githubusercontent.com/4376090/86399348-2476ef00-bcb8-11ea-83c8-814d325786cb.png">
I thought the file names have some similarity.

**Comment by matthewmoss** [comment]
I'm having a similar issue. Did you find a fix for this @ando-khachatryan?

**Comment by ando-khachatryan** [comment]
@matthewmoss  Nope, ended up just deleting those redundant files every time after the simulation.
